Add Simpread extension

Kenshin opened the pull request:
## Description

Search and open [SimpRead Unreader](https://simpread.pro)

## Type of change

- New extension

## Screencast

![2021-11-29_13-35-51](https://user-images.githubusercontent.com/81074/143815341-b26cb8ba-dac9-45f9-8321-85fee9ed3aec.png)

![2021-11-29_13-39-18](https://user-images.githubusercontent.com/81074/143815342-414b37df-3a13-4974-8ab4-a34769bd1fbb.png)

![2021-11-29_13-40-45](https://user-images.githubusercontent.com/81074/143815343-ae525a66-1443-4a73-9f28-0a78d121f3c7.png)

https://user-images.githubusercontent.com/81074/143815314-e6270ced-93e7-4e14-a708-1ce26ba756e1.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

pernielsentikaer reviewed (commented):
Hi @Kenshin 👋 

Thanks for the PR, it looks fantastic 🔥 

Can you maybe record a screencast of the flow since I don't have a way to test it locally, that would help my review a lot 😄

Kenshin commented:
@pernielsentikaer 

Hi,  Video is here

https://user-images.githubusercontent.com/81074/143815314-e6270ced-93e7-4e14-a708-1ce26ba756e1.mp4

and test json file

1. plz download this [simpread_config.json.zip](https://github.com/raycast/extensions/files/7624068/simpread_config.json.zip) and unzip. e.g. `/Users/xxx/Downloads/simpread_config.json` 

2. set this path to Welcome 

![2021-11-29_13-51-53](https://user-images.githubusercontent.com/81074/143815702-f4822899-862a-4814-8354-088237a5bba4.png)

3. <kbd>sr</kbd>  like it's  👇 

![2021-11-29_13-35-51](https://user-images.githubusercontent.com/81074/143815341-b26cb8ba-dac9-45f9-8321-85fee9ed3aec.png)

Done

pernielsentikaer reviewed (changes requested):
Hi @Kenshin 👋 

Thanks for the extension 🔥 , I have a few comments 📄 

- What is simpRead excatly? (to make a better (sub)title) 
- I suggest removing the “SimpRead Config Setting” since it's already possible in settings
- There is [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) upon opening, can you look into this?
- You could store the default favicon in assets folder

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Kenshin commented:
Hi @pernielsentikaer

I update some new soure.

> What is simpRead excatly? (to make a better (sub)title)

Current title is **SimpRead** and description like is snapshot  👇 

![image](https://user-images.githubusercontent.com/81074/144387423-97238f68-8389-49e9-8b2c-7033de5ead26.png)

![iShot2021-12-02 16 32 12](https://user-images.githubusercontent.com/81074/144386452-6c063bdc-a380-415a-b8ad-ae761e4334f1.png)

Next version **add url to simpred**, so title is simpread like github

![image](https://user-images.githubusercontent.com/81074/144387620-ab203a98-f728-4b7b-b7ee-c927c465bb8e.png)

> I suggest removing the “SimpRead Config Setting” since it's already possible in settings

When removed it, user can't be re-set again.  😂 

***

And others is Done, source is here https://github.com/raycast/extensions/pull/435/commits/6bc50e3f335b58a39c2874d60e2e682f5caac059  😁 

- [x] There is empty state flicker upon opening, can you look into this?
- [x] You could store the default favicon in assets folder

pernielsentikaer commented on extensions/simpread/package.json:
```diff
@@ -0,0 +1,51 @@
+    },
+    {
+      "name": "sr",
+      "title": "SimpRead Unreader",
+      "description": "List of SimpRead Unreader",
```
```suggestion
      "title": "Browse Posts",
      "subtitle": "SimpRead",
      "description": "Browse SimpRead Posts",
```
Something like above

pernielsentikaer reviewed (changes requested):
Hi @Kenshin 👋 

It's still possible to set another path within Raycast, see below:
![aryygUufTuRzarWb](https://user-images.githubusercontent.com/38033107/144399042-515d23cb-47e6-4208-9429-69bb12fc2858.png)

I was thinking about setting it like suggested if it match

Kenshin commented:
Hi @pernielsentikaer

All is Done https://github.com/raycast/extensions/pull/435/commits/82e34fbb0d14aed9effd70ea34af673fd76d5061

pernielsentikaer reviewed (changes requested):
Hi @Kenshin 🔥 

Thanks for your contribution, I'm going to accept it and then there will be a final review before shipping 🚢 

Have a nice day 🎄

fe9lix commented:
@Kenshin  @pernielsentikaer Is this ready to be merged? Some remarks: 
- Please use standard code formatting via Prettier
- There are a couple of typos in the README

Kenshin commented:
@fe9lix Hi,  3ks reviews

> There are a couple of typos in the README

What exactly is the error?

> Please use standard code formatting via Prettier

Not every merged project uses Prettier  😂

fe9lix commented:
@Kenshin I've left some inline comments re the typos. As for the formatting: that might be the case, but it's still preferable to have consistent formatting as far as possible; this is why we've added Prettier as default and it will make future changes easier since another author wouldn't need to reformat all of you code (leading to unnecessary diffs in the review).

Kenshin commented:
@fe9lix 

Hi, I added `prettier` to this project. https://github.com/raycast/extensions/pull/435/commits/c19472aab054709bae6e5dccff3a1d1f03287db9

pernielsentikaer commented on extensions/simpread/src/sr.tsx:
```diff
@@ -0,0 +1,190 @@
+    fmtDate = (date: Date) => {
+      const format = (value: any) => (value = value < 10 ? '0' + value : value);
+      return (
+        date.getFullYear() +
+        '年' +
+        format(date.getMonth() + 1) +
+        '月' +
+        format(date.getDate()) +
+        '日 ' +
```
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension

pernielsentikaer commented on extensions/simpread/src/sr.tsx:
```diff
@@ -0,0 +1,190 @@
+    template = `
+# ${unread.title}
+
+> 原文地址 [${host}](${unread.url}) at ${unread.create}
```
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension

pernielsentikaer commented on extensions/simpread/src/sr.tsx:
```diff
@@ -0,0 +1,190 @@
+  }, []);
+
+  if (state.error) {
+    showToast(ToastStyle.Failure, '获取失败，请确认是否存在 simpread_config.json');
```
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension

pernielsentikaer reviewed (changes requested):
Tried to correct some spelling errors in the README, please check and correct the one making sense – also check the other correction about localization.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Kenshin commented:
@pernielsentikaer

Hi, I fixed spelling errors and changed localize. https://github.com/raycast/extensions/pull/435/commits/45080907e1ae764763f34de3fc9af9d04233031a

🙏

Kenshin commented:
@pernielsentikaer

Hi, I fixed spelling errors and changed localize.https://github.com/raycast/extensions/pull/435/commits/45080907e1ae764763f34de3fc9af9d04233031a

pernielsentikaer reviewed (changes requested):
Hi @Kenshin 👋 

Looks like we're close now 💪 The format of the date is not correctly, could you do something like “2021/11-25”

![CzVB1RnM4W2WYNjN](https://user-images.githubusercontent.com/38033107/147536917-5028cbcc-c062-4899-acb0-748f9e71d9df.png)

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Kenshin commented:
@pernielsentikaer

> The format of the date is not correctly, could you do something like “2021/11-25”

Hi, Fixed it. https://github.com/raycast/extensions/pull/435/commits/30158ee0367bbe8e30919f577e27d1f643269053

![image](https://user-images.githubusercontent.com/81074/147537965-844790af-a1cc-4390-91c3-b707f3a9513a.png)

pernielsentikaer reviewed (changes requested):
Hi @Kenshin 👋 

It was actually the date in the bottom that you're formatting I was pointing at, the one in top is from the received data and are allowed since it's from the content. 

![ikmWNxA8pNSdZB4l](https://user-images.githubusercontent.com/38033107/147538895-743ecf38-f12d-4ed3-abd4-4844e89127c7.png)

Sorry for the confusing 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Kenshin commented:
@pernielsentikaer

Sorry, It was my carelessness that wasted your review time, It's okay now. https://github.com/raycast/extensions/pull/435/commits/1147890d6b9d0572e0dc55318ad86ba622f933bb

![image](https://user-images.githubusercontent.com/81074/147539592-f5de236d-d686-4b1a-a215-807f0fdd7998.png)

Kenshin commented:
@pernielsentikaer

Sorry, It was my carelessness that wasted your review time, It's okay now. https://github.com/raycast/extensions/pull/435/commits/1147890d6b9d0572e0dc55318ad86ba622f933bb

![image](https://user-images.githubusercontent.com/81074/147539592-f5de236d-d686-4b1a-a215-807f0fdd7998.png)

Kenshin commented:
@pernielsentikaer

Sorry, It was my carelessness that wasted your review time, It's okay now. https://github.com/raycast/extensions/pull/435/commits/1147890d6b9d0572e0dc55318ad86ba622f933bb

![image](https://user-images.githubusercontent.com/81074/147539592-f5de236d-d686-4b1a-a215-807f0fdd7998.png)

pernielsentikaer reviewed (approved):
Hi @Kenshin 👋 

I'm going to accept it and then will my colleague do a final review before it's finally merged 💎

Kenshin commented:
@pernielsentikaer

Haha, it's been too long, but it's a good thing it's finally done. 👋

Kenshin commented:
@pernielsentikaer @fe9lix 👋

Thanks published and reviews 🙏 This is my New year's gift.
